pkgs/development/python-modules/pyobjc-core/default.nix

```diff
@@ -0,0 +1,133 @@
+    ( grep -RF '/usr/bin/xcrun' || true ; ) | cut -d: -f1 | while IFS= read -r file ; do
+      sed -r "s+/usr/bin/xcrun+$(${lib.getExe pkgs.which} xcrun)+g" -i "$file"
+    done
+    ( grep -RF '/usr/bin/python' || true ; ) | cut -d: -f1 | while IFS= read -r file ; do
+      sed -r "s+/usr/bin/python+$(${lib.getExe pkgs.which} python)+g" -i "$file"
+    done
+  '';
+in
+
+buildPythonPackage rec {
+  pname = "pyobjc_core";
+  version = "10.3.1";
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-sgSoDMwHD5qz+K9COjolpv14fiKFCNAMTDD4rFOLpyA=";
+  };
+  nativeBuildInputs = [
+    xcodebuild
+    cctools
+  ];
+  buildInputs = [
+    objc4
+    apple_libffi
+    Foundation
+    GameplayKit
+    MetalPerformanceShaders
+  ];
+  hardeningDisable = [ "strictoverflow" ]; # -fno-strict-overflow is not supported in clang on darwin
+  NIX_CFLAGS_COMPILE = [ "-Wno-error=deprecated-declarations" ];
+  prePatch = ''
+    # TODO: Make patch for setup.py
+    # ignore the manual include flag for ffi, appears that it needs a very specific ffi from sdk (needs confirmation)
+    substituteInPlace setup.py --replace-fail '"-I/usr/include/ffi"' '#"-I/usr/include/ffi"'
+    # make os.path.exists that can spoil objc4 return True
+    substituteInPlace setup.py --replace-fail 'os.path.join(self.sdk_root, "usr/include/objc/runtime.h")' '"/"'
+
+    # Turn off clang’s Link Time Optimization, or else we can’t recognize (and link) Objective C .o’s:
+    sed -r 's/"-flto=[^"]+",//g' -i setup.py
+    # Fix some test code:
+    grep -RF '"sw_vers"' | cut -d: -f1 | while IFS= read -r file ; do
+      sed -r "s+"sw_vers"+"/usr/bin/sw_vers"+g" -i "$file"
+    done
+
+    # TODO: make a patch for tests
+    substituteInPlace \
+      PyObjCTest/test_nsdecimal.py \
+      --replace-fail "Cannot compare NSDecimal and decimal.Decimal" "Cannot compare NSDecimal and (\\\\w+.)?Decimal"
+    substituteInPlace \
+      PyObjCTest/test_bundleFunctions.py \
+      --replace-fail "os.path.expanduser(\"~\")" "\"/var/empty\""
+    substituteInPlace \
+      PyObjCTest/test_methodaccess.py \
+      --replace-fail "testClassThroughInstance2" \
+      "disable_testClassThroughInstance2"
+  '';
+  preBuild = commonPreBuildDarwinMinVersion;
+  # XXX: We’re turning tests off, because they’re mostly working (0.54% failures among 4,600 tests),
+  # and I don’t have any more time to investigate now (maybe in a Nixpkgs contribution in the future):
+  #
+  # YYY: There were broken tests that relied on home paths and printing format, prePatch phase fixes or disables those
+  #
+  # pyobjc-core> Ran 4600 tests in 273.830s
+  # pyobjc-core> FAILED (failures=3, errors=25, skipped=4, expected failures=3, unexpected successes=1)
+  # pyobjc-core> SUMMARY: {'count': 4600, 'fails': 3, 'errors': 25, 'xfails': 3, 'xpass': 0, 'skip': 4}
+  # pyobjc-core> error: some tests failed
+  #
+  # ZZZ: Sorted out the issues with tests and put them in prePatch
+  checkPhase = ''
+    python3 setup.py test
```

Comment:
That is deprecated and we removed already all usages of running tests like that. Please use pytest.

Reply:
I did not manage to do that, looking into the actual code, the author of the package did some extra stuff to bootstrap unittest

Reply:
Can we use `pytestCheckHook` and `pytestFlagsArray = [ "PyObjCTest" ]`?

Reply:
that's what I tried and it was failing with

```
ERROR PyObjCTest/test_voidpointer.py::TestCase - AttributeError: 'TestCase' object has no attribute 'runTest'. Did you mean:...
ERROR PyObjCTest/test_voidpointer.py::TestVoidPointer - AttributeError: 'TestVoidPointer' object has no attribute 'runTest'. Did yo...
ERROR PyObjCTest/test_weakref.py::TestCase - AttributeError: 'TestCase' object has no attribute 'runTest'. Did you mean:...
ERROR PyObjCTest/test_weakref.py::TestWeakrefs - AttributeError: 'TestWeakrefs' object has no attribute 'runTest'. Did you m...
ERROR PyObjCTest/test_weakref.py::TestObjCWeakRef - AttributeError: 'TestObjCWeakRef' object has no attribute 'runTest'. Did yo...
!!!!!!!!!!!!!!!!!! Interrupted: 438 errors during collection !!!!!!!!!!!!!!!!!!!
====================== 11 warnings, 438 errors in 20.36s =======================
```

`unittestCheckHook` is promising, trying to make it work as well

Reply:
niether unittestCheckHook is easy to make working

Reply:
With unittestCheck Hook errors are different and seem to be related to how tests are set up in the library

https://github.com/ronaldoussoren/pyobjc/blob/master/pyobjc-core/setup.py#L216-L329

Reply:
I've updated the PR with all the comments taken into account, but for running tests with all the guidelines.